src/hiero_sdk_python/contract/contract_id.py

```
@@ -134,6 +142,119 @@ def from_string(cls, contract_id_str: str) -> "ContractId":
+        Args:
+            data (bytes): Protobuf-encoded `ContractID` message.
+
+        Returns:
+            ContractId: Reconstructed ContractId instance.
+
+        Raises:
+            TypeError: If data is not bytes.
+            ValueError: If deserialization fails.
+        """
+        if not isinstance(data, (bytes, bytearray)):
+            raise TypeError("data must be bytes")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)